### PR TITLE
Leader Election Metrics 1a: Client Side Wiring

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
* Add metrics to the client to monitor the perceived impact of leader elections. Note that further metrics on the actual length of leader elections will be introduced in https://github.com/palantir/atlasdb/pull/5069.

**Implementation Description (bullets)**:
* Extracted an interface for `NamespacedConjureTimelockService`, and created another class that delegates to the implementation.
* Monitor startTransactions and getCommitTimestamp requests, and report metrics when these perceive a leader change.

**Testing (What was existing testing like?  What have you done to improve it?)**:
* Added a unit test; intend on adding ETE tests later (in a followup PR).

**Concerns (what feedback would you like?)**:
* Confirm that this won't affect performance

**Where should we start reviewing?**:
* `LeaderElectionReportingTimelockService`

**Priority (whenever / two weeks / yesterday)**:
This week

### CODE REVIEW

File: lock-api/src/main/java/com/palantir/lock/client/RemoteTimelockServiceAdapter.java [hunk 1]
```diff
@@ -59,17 +57,6 @@ public static RemoteTimelockServiceAdapter create(
         return new RemoteTimelockServiceAdapter(rpcClient, conjureClient, lockWatchEventCache);
     }
 
-    public static RemoteTimelockServiceAdapter create(
```
Comment (Jolyon-S) [inline lock-api/src/main/java/com/palantir/lock/client/RemoteTimelockServiceAdapter.java:62]: drive-by

**Review by gmaretic — APPROVED** [review]
We paired on this so I'm good with it, but going to leave here so others have a chance to look through if they like.

This should be completely not contentious except if someone has a better idea how to wire it into `TimeLockFeedbackBackgroundTask`

File: lock-api/src/main/java/com/palantir/lock/client/NamespacedConjureTimelockService.java [hunk 2]
```diff
@@ -24,54 +24,27 @@
...
-import com.palantir.tokens.auth.AuthHeader;
 
-public class NamespacedConjureTimelockService {
-    private static final AuthHeader AUTH_HEADER = AuthHeader.valueOf("Bearer omitted");
-    private final String namespace;
-    private final ConjureTimelockService conjureTimelockService;
+public interface NamespacedConjureTimelockService {
+    ConjureUnlockResponse unlock(ConjureUnlockRequest request);
 
-    public NamespacedConjureTimelockService(ConjureTimelockService conjureTimelockService, String namespace) {
```
Comment (jeremyk-91) [inline lock-api/src/main/java/com/palantir/lock/client/NamespacedConjureTimelockService.java:41]: This will have a number of downstream breaks in the internal backup product and internal timelock migration CLI tools. Overall impact is limited, but we need to ensure these get fixed.

**Review by jeremyk-91 — APPROVED** [review]
Broadly looks good. Please make sure to track fixing the services that will be broken by this change
